CC module consistency checking

amjames opened the pull request:
## Description
Adds regression tests which ensure consistency between different `qc_module`s for `cc_type`= `conv`. Should finish off #714?

## Questions
- [x] What lists should this test be on? Currently listed as `psi;quicktests;cc` because I copied and minimally modified from `CMakeLists.txt` in the mp2-module test. 
- [x]  Managed methods [table](http://psicode.org/psi4manual/master/notes_c.html#table-managedmethods) lists DETCI as a possible choice for conventional CCSD with RHF/ROHF reference, but I errors from the driver trying these combinations.
```python
set qc_module DETCI
energy('ccsd')

RuntimeError:
FatalError: DETCI: The CC module is not currently enabled. 
```
## TODO
 - [x] ~Add ROHF/RHF CCSD checks against DETCI if they apply (pending feedback about 2nd point above)~ Remove indications that DETCI can do CCSD from the table
 - [x] Add checks between fnocc and dfocc
 - [x] Add non-frozen-core tests

## Status
- [x] Ready for review
- [x] Ready for merge

loriab commented:
Could you test non-frozen-core, too? And maybe DF between fnocc & dfocc?

dgasmith commented:
The CC module for DETCI was never ported over from Psi3, feel free to remove references that DETCI can do this.

loriab commented:
If you would, please. Sanity checks never hurt. Anything that's doubled in `select_ccsd()`. Can remove https://github.com/psi4/psi4/blob/master/psi4/driver/procrouting/proc.py#L721-L722 and https://github.com/psi4/psi4/blob/master/psi4/driver/procrouting/proc.py#L743-L744 along with the detci docs.

amjames commented:
Removed the DETCI refs from the docs table, and in select ccsd. 

I also updated occ psivars.  It was setting `CURRENT CORRELATION ENERGY` for `{df/cd}-ccsd(t)` but it was not setting `CCSD(T) CORRELATION ENERGY`.

The test now runs through a more complete set of cross checking. Testing conv (fnocc, ccenergy) and df/cd (fnocc, occ) both with and without frozen core. 

Note: The changes to occ required a rebuild, and I am noticing when doing df/cd (either occ or fnocc module) I get a message printed to `stdout`"No atoms detected, returning". Are we aware of this?

amjames commented:
Also waiting on input about where to put this test, with the larger set of checks it is most too long for  `quicktests`

amjames commented on tests/cc-module/CMakeLists.txt:
```diff
@@ -0,0 +1,3 @@
+include(TestingMacros)
+
+add_regression_test(cc-module, "psi;quicktests;cc")
```
Running on 1 core it takes 10s to run my test, and just checking you want me to add the line
`set_tests_properties(python-vibanalysis PROPERTIES COST 700)` to the vibanalysis test? 

Do you want me to add anything like that to the test from this PR? 

What does that do anyway can `ctest` filter tests by cost?

loriab replied:
It runs 12 CC jobs in 10s? Good for it!

Yes, that looks right on the `COST` line. Yeah, you will have noticed that `ctest` runs tests in order by cost _after_ the first time in an objdir. This just gives it a head start for the first time. And no, we're only adding `COST` on tests >100s.

Go ahead and shift the "No atoms detected, returning" lines, too, if you would.

amjames commented:
Okay all my questions have been answered, so pending reviews this is ready to merge.

I added the cost property to the vibanalysis test and applied the fix in `qcdb/molecule.py` to avoid the "No atoms detected, returning" lines printing.

Since the test runs in 10s I left it in quicktests.

Last question to answer, does this close out #714 or should that issue stay open after merge?

loriab commented:
I think it closes #714
